h2/connection.py

```diff
@@ -858,25 +858,35 @@ def send_data(self, stream_id, data, end_stream=False):
         :type data: ``bytes``
         :param end_stream: (optional) Whether this is the last data to be sent
             on the stream. Defaults to ``False``.
+        :param pad_length: (optional) Length of the padding to apply to the
+            data frame. Defaults to ``None`` for no use of padding. Note that
+            a value of ``0`` results in padding of length ``0``
+            (with the "padding" flag set on the frame).
         :type end_stream: ``bool``
         :returns: Nothing
         """
-        if len(data) > self.local_flow_control_window(stream_id):
+        frame_size = len(data)
+        if pad_length is not None:
+            frame_size += pad_length + 1
```

Comment:
So, I know why we need the `+1`, but it's still worth a comment expressly because it's unclear unless you've read the RFC fairly closely.

(@alexwlchan an FYI, the reason is that when you add the PADDING flag you don't just add the padding bytes, you also add an extra byte that indicates how many padding bytes there are. That byte is also part of the flow controlled length.)